extensions/resteasy-common/runtime/src/main/java/io/quarkus/resteasy/common/runtime/ResteasyContextProvider.java

```diff
@@ -13,7 +14,10 @@
 
     @Override
     public ThreadContextSnapshot currentContext(Map<String, String> props) {
-        Map<Class<?>, Object> context = ResteasyContext.getContextDataMap();
+        Map<Class<?>, Object> context = ResteasyContext.getContextDataMap(false);
+        if (context == null) {
+            return null;
+        }
```

Comment:
Not sure this works, because you may need to capture the fact that there's no context, to make sure that when you run the action you remove any existing context.

Reply:
Does that actually matter? There won't be a context unless you are inside a RESTEasy request, or unless you have saved it.

Reply:
Well, you don't know where a `Uni` is coming from. If it came from a place outside of a request context, it will have no RESTEasy context. If you then complete it from within a context, it will inherit the RESTEasy context by accident, where it should not.

Reply:
I still don't think this is a real issue. If there is a RESTEasy request is progress then CP will have captured the context, so it should be propagated everywhere and there is no change of behaviour. The only way this will be missing is if a request is started with no request is progress, and then ends up in a RESTEasy request, such as with lazy authentication. This could mean that RESTEasy requests effectively loose their context once they start dealing with the SecurityIdentity, which does not seem ideal IMHO.